File: app/templates/app-java/src/main/java/domain/executors/ThreadExecutor.java
package <%= appPackage %>.domain.executors;

import java.util.concurrent.Callable;
import java.util.concurrent.Executor;
import java.util.concurrent.Future;

public interface ThreadExecutor extends Executor {

    Future<?> submit(Runnable task);

    <T> Future<T> submit(Callable<T> task);

}

File: app/templates/app-kotlin/src/main/java/domain/executors/ThreadExecutor.kt
package <%= appPackage %>.domain.executors;

import java.util.concurrent.Callable;
import java.util.concurrent.Executor;
import java.util.concurrent.Future;

interface ThreadExecutor : Executor {

  fun submit(task: Runnable): Future<*>

  fun <T> submit(task: Callable<T>): Future<T>

}
